### PR TITLE
adjust some parameters for REACH data (H band)

### PR DESCRIPTION
**Pull request by YuiKasagi**
I made some changes to a few files for REACH data.
**major change**
- `mask.py`, `oned_extract.py`: Adjust trace width. (because of this change, some tests are modified)
- `REACH_stream.py`: Almost the same with IRD_stream, but the branch of instrument change the trace width.

**minor change**
- `aptrace.py`: Improve the peak detection method
-  `wavcal.py`: Available to use weight when fitting. (this change also cause the change of a test file)

There remains some issues need to be considered...
- This PR may make any improvement for the accuracy of the wavelength calibration because the weights added in `wavcal.py` are not used yet. I will update this after some more verification. (But the changes for the trace width can affect the accuracy (?))
- Need to adjust for REACH YJ band data

### CODE REVIEW

**Comment by HajimeKawahara** [comment]
Thanks. All of the unit tests passed.

Can you add 
`include src/pyird/data/IP_fwhms_h.dat`
in MANIFEST.in?

**Comment by kemasuda** [comment]
Thanks for the updates. The changes look fine to me and the tests all worked in my environment. One thing I realized is that the targets were in star fiber in the 20211110_REACH data for some reason, so the new REACH_stream.py fails for these data; the spectrum looks fine if (i) mmf2 is used and (ii) ThAr spectra taken with star fiber (53347-53410) are used. Sorry about this confusion.

**Comment by kemasuda** [comment]
This itself is not an issue of the pipeline, though I realized that the reduction using the updated pipeline (assuming that the above modification is correct) resulted in a larger telluric RV shifts than before (see figures; this is for the frame 53205). So the bias in the wavelength calibration seems related to the aperture size. 
telluric_rvs
72649/182820624-e4359883-e473-4ae1-9c53-1449e33ee989.png)

**Comment by YuiKasagi** [comment]
Thank you for sharering these interesting results. I didn’t realized that changing aperture size have such a large effect for the wavelength accuracy. It should be considered for the next PR...

File: src/pyird/utils/irdstream.py [hunk 1]
```diff
@@ -281,7 +302,13 @@ def calibrate_wavelength(self, trace_file_path=None, maxiter=30, stdlim=0.001, n
                 median_image, trace_legendre, y0, xmin, xmax, coeff)
             rsd = multiorder_to_rsd(rawspec, pixcoord)
             print(np.shape(rsd.T)[0])
-            wavsol, data = wavcal_thar(rsd.T, maxiter=maxiter, stdlim=stdlim)
+            ## set weights
+            if self.band=='h':
+                #w = make_weight()
+                w = np.ones(rsd.shape) #XXX!!
```
Comment (YuiKasagi) [inline src/pyird/utils/irdstream.py:308]: memo:
One can set the weight of the fitting ThAr emissions in the wavelength calibration here.

**Comment by YuiKasagi** [comment]
Thank you for reviewing! I committed again after editing `MANIFEST.in` (and `REACH_stream.py` since I also checked that the reduction works for the target data).

File: src/pyird/utils/irdstream.py (hunk 1 above)
Comment (kemasuda) [reply src/pyird/utils/irdstream.py:308]: Yeah I don't know why the aperture size is related, hopefully this helps to resolve the inaccuracy in future PRs... 
I also tried using w from make_weight(); accuracy of the middle orders may have been slightly improved but those at the edges (i.e. 1,2 or 15,16) didn't change, so the origin might be different. Anyway thanks a lot for these updates!

**Review by kemasuda — APPROVED** [review]

File: src/pyird/utils/irdstream.py (hunk 1 above)
Comment (YuiKasagi) [reply src/pyird/utils/irdstream.py:308]: Thank you for trying to use make_weight. But it doesn't make significant improvement, I see...

**Review by HajimeKawahara — APPROVED** [review]
Thanks. I approve it.

**Pull request merged**